oozie-to-airflow/mappers/archive_file_mixin.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Dict
+
+from mappers.file_mixin import FileMixin
+
+ALLOWED_EXTENSIONS = [".zip", ".gz", ".tar.gz", ".tar", ".jar"]
+
+
+class ArchiveAndFileMixin(FileMixin):
```

Comment:
Maybe we could rethink the inheritance structure a bit here.
First of all, I'm not a fan of a class having an `And` in the name - that's a "code smell" for me, indicating that there's probably a slightly better way of structuring the code.
So my thinking here is: what if we extracted the `_check_path_for_comma()` and `_split_by_hash_sign()` methods from `FileMixin` to a utility class (not sure what the name might be, so calling it `GummyBear` for now, because why not) which we could then use by composition
```
gb = GummyBear()
```

This way I think we could have `ArchiveMixin` and `FileMixin` which would be completely independent of each other, both including `GummyBear` by composition and both following SRP. Note: `ArchiveMixin` would no longer inherit from `FileMixin`.

I believe this way the code could be a bit "cleaner" and if a mapper needs both mixins, it would just inherit from both `MyMapper(ArchiveMixin, FileMixin)`. This approach also would follow the Interface Segregation Principle.

What do you think?

Reply:
Agree. Will do.